universal-application-tool-0.0.1/test/controllers/HomeControllerAuthenticatedTest.java

```diff
@@ -0,0 +1,50 @@
+package controllers;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static play.api.test.Helpers.testServerPort;
+import static play.test.Helpers.*;
+
+import com.google.common.collect.ImmutableMap;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.pac4j.core.client.Clients;
+import org.pac4j.core.client.direct.AnonymousClient;
+import org.pac4j.core.config.Config;
+import org.pac4j.core.context.HttpConstants;
+import play.Application;
+import play.mvc.Http;
+import play.mvc.Result;
+import play.test.WithApplication;
+
+public class HomeControllerAuthenticatedTest extends WithApplication {
+
+  // Should be no need for a database here.
+  protected Application provideApplication() {
+    ImmutableMap<String, String> args =
+        new ImmutableMap.Builder<String, String>()
+            .putAll(inMemoryDatabase("default", ImmutableMap.of("MODE", "PostgreSQL")))
+            .put("play.evolutions.db.default.enabled", "false")
+            .build();
+    return fakeApplication(args);
+  }
+
+  @Before
+  public void setUp() {
+    Config config = app.injector().instanceOf(Config.class);
```

Comment:
question: Are we sure this is using the `fakeApplication` above? It's not clear to me that provideApplication() is actually being called

Reply:
Yep!  That's what WithApplication does.  Sets it as the variable `app`.

Reply:
Ah good to know, it's messing me up that there's no @Override for that annotation. The same happens with the Ebean config (no override needed for startup)